### ISSUE TITLE
exercises 05a & 05b

### ISSUE THREAD

**Issue by dimitrilw** [dbusteed/mojolings#4]
I think there would be value in showing how variables are scoped. Something like:

```mojo
fn main():
    # this "x" is initialized at the top level
    var x = "x1" 
    print("first top-level print:", x)

    # when entering a new block of code (most indent levels are blocks),
    # then variables initialized inside the block will be dropped at the end of the block
    for n in range(2):
        # this "x" is initialized inside the for-loop; it has an inner-scope
        var x = "x2 " + str(n)
        print("for-loop print:", x)
        # the for-loop is closing; variables scoped to it (like the inner x) will be dropped

    print("second top-level print:", x)

    if x == "x1":
        print("x is the top-level scope")
    else:
        print("x is the inner scope")
```

The output of that program:
```
first top-level print: x1
for-loop print: x2 0
for-loop print: x2 1
second top-level print: x1
x is the top-level scope
```

The lesson commentary can make a point about the importance of unique variable names (i.e., repeated use of "x" is a code smell) and also pose the question of "what if you wanted the variable to be populated inside the for-loop?" Which is answered in the next exercise, currently exercise 05.

Granted, that trite code snip above should be redone to not just demonstrate a concept, but also create a fail-state.

Thoughts?

**Comment by dbusteed** [comment]
I agree variable scoping should be addressed with a unique exercise.

I'm pretty open-minded on the ordering of exercises, although I like the general idea of breadth before depth, i.e. learn the basics of variables, functions, etc, then in a later block learn the more complicated features of variables, functions, etc. But I imagine scoping is a concept that should be covered sooner rather than later.

On a somewhat related note, I'm thinking about splitting the `02-var-vs-alias` into two, and introduce the `alias` keyword after other basics.

**Comment by dimitrilw** [comment]
I think breaking var & alias apart is a really good idea. One of the nice things about rustlings is that it is soooo granular. It is like the Duolingo of learning Rust. You don't even feel like you're learning, but then a few exercises later -- *poof*, you learned stuff. Tiny digestible little lessons.
